File: code examples/golang/amazon_bestsellers.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	const Username = "YOUR_USERNAME"
	const Password = "YOUR_PASSWORD"

	payload := map[string]interface{}{
		"source":     "amazon_bestsellers",
		"domain":     "com",
		"query":      "Clothing, Shoes & Jewelry",
		"start_page": 2,
		"parse":      true,
		"context": []map[string]interface{}{
			{"key": "category_id", "value": 6127770011},
		},
	}

	jsonValue, _ := json.Marshal(payload)

	client := &http.Client{}
	request, _ := http.NewRequest("POST",
		"https://realtime.oxylabs.io/v1/queries",
		bytes.NewBuffer(jsonValue),
	)

	request.SetBasicAuth(Username, Password)
	response, _ := client.Do(request)

	responseText, _ := ioutil.ReadAll(response.Body)
	fmt.Println(string(responseText))
}

File: code examples/python/amazon_bestsellers.py
import requests
from pprint import pprint


# Structure payload.
payload = {
    'source': 'amazon_bestsellers',
    'domain': 'com',
    'query': 'Clothing, Shoes & Jewelry',
    'start_page': 2,
    'parse': True,
    'context': [
        {'key': 'category_id', 'value': 6127770011},
    ],
}


# Get response.
response = requests.request(
    'POST',
    'https://realtime.oxylabs.io/v1/queries',
    auth=('user', 'pass1'),
    json=payload,
)

# Print prettified response to stdout.
pprint(response.json())